app/assets/javascripts/search_dashboard.js

```diff
@@ -0,0 +1,125 @@
+search = angular.module('pcbuilder', []);
+
+search.controller("searchController", ['$scope', '$http', function ($scope, $http) {
+  var pathName = window.location.pathname;
+  var categories;
+  $http.get("/api/dashboard/category").success(function (category) {
+    categories = category.categories;
+    $http.get("/api"+pathName).success(function (data) {
+      $scope.products = data;
+    });
+  });
+  
+  $scope.submitform = function () {
+    window.location.href = $scope.searchTag;
+  }
+  $scope.getCategory = function (msg) {
+    for(var i = 0; i <= categories.length ; i ++){
+      if(i == categories.length){
+        if(msg == 'Processors'){
+          return 'cpu';
+        }else if(msg == 'CPU-koelers'){
```

Comment:
Add a locale in the database, don't check for it.  Also all categories do contain a locale now